cfnstack/cfnstack.go

```diff
@@ -50,3 +59,61 @@ func StackEventErrMsgs(events []*cloudformation.StackEvent) []string {
 
 	return errMsgs
 }
+
+func NestedStackExists(cf CFInterrogator, parentStackName, stackName string) (bool, error) {
+	logger.Debugf("Testing whether nested stack '%s' is present in parent stack '%s'", stackName, parentStackName)
+	parentExists, err := StackExists(cf, parentStackName)
+	if err != nil {
+		return false, err
+	}
+	if !parentExists {
+		logger.Debugf("Parent stack '%s' does not exist, so nested stack can not exist either", parentStackName)
+		return false, nil
+	}
+
+	req := &cloudformation.ListStackResourcesInput{StackName: &parentStackName}
+	logger.Debugf("Calling AWS cloudformation ListStackResources for stack %s ->", parentStackName)
+	out, err := cf.ListStackResources(req)
+	if err != nil {
+		return false, fmt.Errorf("Could not read cf stack %s: %v", parentStackName, err)
+	}
+	if out == nil {
+		return false, nil
+	}
+	logger.Debugf("<- AWS responded with %d stack resources", len(out.StackResourceSummaries))
+	for _, resource := range out.StackResourceSummaries {
+		if *resource.LogicalResourceId == stackName {
+			logger.Debugf("Match! resource id '%s' exists", stackName)
+			return true, nil
+		}
+	}
+	logger.Debugf("No match! resource id '%s' does not exist", stackName)
+	return false, nil
+}
+
+func StackExists(cf CFInterrogator, stackName string) (bool, error) {
```

Comment:
Would you mind making this a private func `stackExists` in case this isn't called outside of the pkg?

Reply:
Ah yes, I see what you mean - yes it isn't used now but it feels like the sort of utility function that will be used in itself very soon, e.g. we could use it to implement a kube-aws apply command (that could replace separate up and update)?

Reply:
> utility function that will be used in itself very soon, e.g. we could use it to implement a kube-aws apply command

Yeah, that makes sense! I'd love `kube-aws apply` after `kubectl apply` and `terraform apply`.

However, even in that case what we'd need is `func (c *Provisioner) createOrUpdateStackWithTemplateURL(...)` which calls `stackExists` in order to decide it is a create or an update, so we'd better make it private within the `cfnstack` pkg? 🤔

Reply:
Please see my latest commit with a proposed fail-fast on update when the root stack or nested etcd stack doesn't exist.  The root stack does not exist is best served with the StackExists function as we want to deal with this more generic case independently before checking the missing nested stack case.